File: Backend/Models/Transacao.cs
namespace Backend.Models;

public class Transacao
{
    public enum TipoDeTransacao
    {
        Receita = 0,
        Despesa = 1 //O codigo manda apenas 0 ou 1 evitado conflito de nomes "diferentes": Despes, DESPESA, despesa.
    }

    public int TransacaoId {get; set;}
    public string Descricao {get; set;} = string.Empty; //evitar começar como Null
    public decimal Valor {get; set;}
    public TipoDeTransacao Tipo {get; set;}
    public int PessoaId {get; set;}// a "chave estrangeira" (guarda só o número)

    
    public Pessoa Pessoa { get; set; } = null!; // a "propriedade de navegação" (o objeto inteiro)

}

File: Frontend/src/types/Transacao.ts
import type { Pessoa } from "./Pessoa.ts";

export type TipoDeTransacao = 0 | 1;

export const TipoDeTransacaoEnum = {
  Receita: 0,
  Despesa: 1,
} as const;

export interface Transacao {
  transacaoId: number;
  descricao: string;
  valor: number;
  tipo: TipoDeTransacao;
  pessoaId: number;
  pessoa: Pessoa; // vem preenchida por causa do .Include(t => t.Pessoa) no backend
}

export interface CreateTransacaoDto {
  descricao: string;
  valor: number;
  tipo: TipoDeTransacao;
  pessoaId: number;
}